Full Store Editing: Technical documentation (#5939)
woocommerce/woocommerce-gutenberg-products-block: assets/js/blocks/legacy-template/README.md

> # Legacy Template Block

Reviewer: [Since we are renaming those to Classic Template Blocks](https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/5861), even though that issue is not closed yet, I'd suggest we already call them like that in the docs, since these docs are not going to be out before the rename anyways.

Author: Yeah, that is fine. I was unsure on whether to use the term Legacy or Classic when writing these docs so I kept it inline with what was already there (which happened to be Legacy). I will update all references to say Classic.

The Legacy Template block is a placeholder block for specific WooCommerce block templates which are rendered on the server-side when a block theme is active.

> Given a specific template attribute prop, the block will render a specific template both on the front-end and in the Site Editor in the form of a placeholder.

Reviewer: I feel this sentence is a bit ambiguous, but always keep in mind that English is not my native language. I find it ambiguous on two fronts:

> Given a specific template attribute prop, the block will render a specific template

This is a bit unclear more than ambiguous. What about something like:

_By assigning a template identifier to the attribute prop, the block will render that specific template_

> both on the front-end and in the Site Editor in the form of a placeholder.

Here I feel that it sounds like the placeholder will be rendered both on the front-end and in the Site Editor. Perhaps it's because of “both”? Maybe a comma between “front-end” and “and” could help? Or perhaps, the entire sentence could be something like this:

_By assigning a template identifier to the attribute prop, the block will render that specific template on the front-end, and a placeholder for said template in the Site Editor_

> 

Reviewer: I'm wondering if here we should add a paragraph explaining that the placeholder does not completely represent what is rendered on the front-end.

[Considered that a suggestion](pdnLyh-tW-p2) by @nerrad  was to apply some styles to the template placeholder eventually, this might be useful so that people are aware that this is not going to happen yet and don't get surprised when it doesn't.

What do you think?

Author: >I'm wondering if here we should add a paragraph explaining that the placeholder does not completely represent what is rendered on the front-end.

It cannot hurt to add a paragraph explaining that so no problem, it's probably more useful to belong in the user/merchant facing docs but either way not a problem.

>this might be useful so that people are aware that this is not going to happen yet and don't get surprised when it doesn't.

Personally, I don't think this really belongs in these docs. Once a technical decision & change is made we should also update the documentation accordingly. I'd prefer to avoid adding "X or Y might be coming in the future" as it can lead to some confusion and muddy the waters for those who are looking for what is implemented right now.

We can however refer to other documentation, or a roadmap where this might be more applicable?

Reviewer: Yea, I just mean we should warn people that styles and other customizations won't be applied to the placeholder. I'm not suggesting we necessarily write in the docs that it's our intention to make that happen eventually.

(Though, to be fair, we have some reference to future plans [here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/5939/files#diff-162dd54970925f7e59d7d2b0fdda4785f6b4ed538b628459e4a89bf2f7767952R43), though I think it's different in this case since that's an over-arching goal and not a detail such as styles applied to the placeholder, so I'd be for leaving that reference to the long-term blockification into the docs, personally)

Author: >Yea, I just mean we should warn people that styles and other customizations won't be applied to the placeholder.

I see what you mean. At the moment the block does not have any customizable options anyway, but I'll make a note of it anyway in the docs.

# Classic Template Block

The Classic Template block is a placeholder block for specific WooCommerce block templates which are rendered on the server-side when a block theme is active.

By assigning a template identifier to the attribute prop, the block will render that specific template on the front-end, and a placeholder for said template in the Site Editor.

> It's worth noting that in the Site Editor where a placeholder merely represents a front-end view, this representation will not always be completely accurate.

Reviewer 2: ```suggestion
It's worth noting that the placeholder in the Site Editor is merely an approximate representation of a template's front-end view.
```
